sunpy/map/mapbase.py

```diff
@@ -1142,8 +1194,11 @@ def submap(self, range_a, range_b):
             if range_b[1] is None:
                 range_b[1] = self.yrange[1]
 
-            x1, y1 = np.ceil(u.Quantity(self.data_to_pixel(range_a[0], range_b[0]))).value
-            x2, y2 = np.floor(u.Quantity(self.data_to_pixel(range_a[1], range_b[1])) + 1*u.pix).value
+            coord1 = SkyCoord(range_a[0], range_b[0], frame=self.coordinate_frame)
+            coord2 = SkyCoord(range_a[1], range_b[1], frame=self.coordinate_frame)
+
+            x1, y1 = np.ceil(u.Quantity(self.data_to_pixel(coord1))).value
+            x2, y2 = np.floor(u.Quantity(self.data_to_pixel(coord2)) + 1*u.pix).value
```

Comment:
which I think means you never end up indexing out a dimension (i.e. you always have at least one pixel)